File: 832-flipping-an-image/832-flipping-an-image.js
/**
 * @param {number[][]} image
 * @return {number[][]}
 */
var flipAndInvertImage = function(image) {
    let arr = [];
    for(let i=0; i<image.length; i++){
        let bag = '';
        // for(let j=0; j<image[i].length; i++){
        for(let j=image[i].length-1; j>=0; j--){
            // arr[i][(image[i].length-1)-j] = image[i][j];   
            bag = bag + image[i][j];
        }
        let temp = bag.split("");
        // console.log(temp);
        arr.push(temp);
    }
    for(let i=0; i<arr.length; i++){
        for(let j=0; j<arr[i].length; j++){
            if(arr[i][j] == 1){
                arr[i][j] = 0;
            }else{
                arr[i][j] = 1;
            }
        }
    }
    return arr
};

File: 832-flipping-an-image/832-flipping-an-image.java
class Solution {
    public int[][] flipAndInvertImage(int[][] image) {
        int[][] ans = new int[image.length][image[0].length];
        // System.out.println(Arrays.toString(image));
        for(int i=0; i<image.length; i++){
            // for(int j=0; j<image[i].length; j++){  
            for(int j=image[i].length-1; j>=0; j--){
                ans[i][image[i].length-j-1] = image[i][j];
            }
        }
        // System.out.println(Arrays.toString(ans));
        for(int i=0; i<image.length; i++){
            for(int j=0; j<image.length; j++){
                if(ans[i][j] == 0){
                    ans[i][j] = 1;
                }else{
                    ans[i][j] = 0;
                }
            }
        }
        return ans;
    }
}